azkaban-common/src/test/java/azkaban/executor/ExecutorManagerTest.java

```diff
@@ -0,0 +1,303 @@
+        new Executor[] { executor1 });
+
+      // mark older executor as inactive
+      executor1.setActive(false);
+      loader.updateExecutor(executor1);
+      manager.setupExecutors();
+      Assert.fail("Expecting exception, but didn't get one");
+    } catch (ExecutorManagerException ex) {
+      System.out.println("Test true");
+    }
+  }
+
+  /* Test disabling queue process thread to pause dispatching */
+  @Test
+  public void testDisablingQueueProcessThread() throws ExecutorManagerException {
+    ExecutorManager manager = createMultiExecutorManagerInstance();
+    manager.enableQueueProcessorThread();
+    Assert.assertEquals(manager.isQueueProcessorThreadActive(), true);
+    manager.disableQueueProcessorThread();
+    Assert.assertEquals(manager.isQueueProcessorThreadActive(), false);
+  }
+
+  /* Test renabling queue process thread to pause restart dispatching */
+  @Test
+  public void testEnablingQueueProcessThread() throws ExecutorManagerException {
+    ExecutorManager manager = createMultiExecutorManagerInstance();
+    Assert.assertEquals(manager.isQueueProcessorThreadActive(), false);
+    manager.enableQueueProcessorThread();
+    Assert.assertEquals(manager.isQueueProcessorThreadActive(), true);
+  }
+
+  /* Test submit a non-dispatched flow */
+  @Test
+  public void testQueuedFlows() throws ExecutorManagerException, IOException {
+    ExecutorLoader loader = new MockExecutorLoader();
+    ExecutorManager manager = createMultiExecutorManagerInstance(loader);
+    ExecutableFlow flow1 = createExecutableFlow("exec1");
+    flow1.setExecutionId(1);
+    ExecutableFlow flow2 = createExecutableFlow("exec2");
+    flow2.setExecutionId(2);
+
+    User testUser = getTestUser();
+    manager.submitExecutableFlow(flow1, testUser.getUserId());
+    manager.submitExecutableFlow(flow2, testUser.getUserId());
+
+    List<ExecutableFlow> testFlows = new LinkedList<ExecutableFlow>();
+    testFlows.add(flow1);
+    testFlows.add(flow2);
+
+    List<Pair<ExecutionReference, ExecutableFlow>> queuedFlowsDB =
+      loader.fetchQueuedFlows();
+    Assert.assertEquals(queuedFlowsDB.size(), testFlows.size());
+    // Verify things are correctly setup in db
+    for (Pair<ExecutionReference, ExecutableFlow> pair : queuedFlowsDB) {
+      Assert.assertTrue(testFlows.contains(pair.getSecond()));
+    }
+
+    // Verify running flows using old definition of "running" flows i.e. a
+    // non-dispatched flow is also considered running
+    List<ExecutableFlow> managerActiveFlows = manager.getRunningFlows();
+    Assert.assertTrue(managerActiveFlows.containsAll(testFlows)
+      && testFlows.containsAll(managerActiveFlows));
+
+    // Verify getQueuedFlowIds method
+    Assert.assertEquals("[1, 2]", manager.getQueuedFlowIds());
+  }
+
+  /* Test submit duplicate flow when previous instance is not dispatched */
+  @Test
+  public void testDuplicateQueuedFlows() throws ExecutorManagerException,
```

Comment:
expecting an exception?
http://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests